src/fastoad/utils/postprocessing/analysis_and_plots.py

```diff
@@ -343,246 +361,119 @@ def mass_breakdown_sun_plot(aircraft_file_path: str, file_formatter=None):
-                "MTOW" + "<br>" + str(int(MTOW)) + " [kg]",
+                "MTOW" + "<br>" + str(int(mtow)) + " [kg]",
                 "payload"
                 + "<br>"
                 + str(int(payload))
                 + " [kg] ("
-                + str(round(payload / MTOW * 100, 1))
+                + str(round(payload / mtow * 100, 1))
                 + "%)",
                 "fuel_mission"
                 + "<br>"
                 + str(int(fuel_mission))
                 + " [kg] ("
-                + str(round(fuel_mission / MTOW * 100, 1))
+                + str(round(fuel_mission / mtow * 100, 1))
                 + "%)",
-                "OWE" + "<br>" + str(int(OWE)) + " [kg] (" + str(round(OWE / MTOW * 100, 1)) + "%)",
+                "OWE" + "<br>" + str(int(owe)) + " [kg] (" + str(round(owe / mtow * 100, 1)) + "%)",
             ],
             parents=[
                 "",
-                "MTOW" + "<br>" + str(int(MTOW)) + " [kg]",
-                "MTOW" + "<br>" + str(int(MTOW)) + " [kg]",
-                "MTOW" + "<br>" + str(int(MTOW)) + " [kg]",
+                "MTOW" + "<br>" + str(int(mtow)) + " [kg]",
+                "MTOW" + "<br>" + str(int(mtow)) + " [kg]",
+                "MTOW" + "<br>" + str(int(mtow)) + " [kg]",
             ],
-            values=[MTOW, payload, fuel_mission, OWE],
+            values=[mtow, payload, fuel_mission, owe],
             branchvalues="total",
         ),
         1,
         1,
     )
 
-    airframe_str = (
-        "airframe"
-        + "<br>"
-        + str(int(airframe))
-        + " [kg] ("
-        + str(round(airframe / OWE * 100, 1))
-        + "%)"
-    )
-    propulsion_str = (
-        "propulsion"
-        + "<br>"
-        + str(int(propulsion))
-        + " [kg] ("
-        + str(round(propulsion / MTOW * 100, 1))
-        + "%)"
-    )
-    systems_str = (
-        "systems"
-        + "<br>"
-        + str(int(systems))
-        + " [kg] ("
-        + str(round(systems / MTOW * 100, 1))
-        + "%)"
-    )
-    furniture_str = (
-        "furniture"
-        + "<br>"
-        + str(int(furniture))
-        + " [kg] ("
-        + str(round(furniture / MTOW * 100, 1))
-        + "%)"
-    )
-    crew_str = (
-        "crew" + "<br>" + str(int(crew)) + " [kg] (" + str(round(crew / MTOW * 100, 1)) + "%)"
-    )
-
+    # Get data:weight 2-levels decomposition
+    categories_values = []
+    categories_names = []
+    categories_labels = []
+    for variable in variables.names():
+        name_split = variable.split(":")
+        if isinstance(name_split, list) and len(name_split) == 4:
+            if name_split[0] + name_split[1] + name_split[3] == "dataweightmass" and not (
+                "aircraft" in name_split[2]
+            ):
+                categories_values.append(
+                    convert_units(variables[variable].value[0], variables[variable].units, "kg")
+                )
+                categories_names.append(name_split[2])
```

Comment:
duplication avec lignes 331-340.

Reply:
Corrected in a23939f

Reply:
C'est là que je ne vois pas de changement.

Reply:
Oups pardon j'ai cru que c'était lié à la remarque précédente. C'était pas exactement similaire mais j'espère que ce que je propose te va ;) 
Corrigé dans 54f43d2